### PR TITLE
migrate_options_shared: Add new case for mPHB migration

### PR DESCRIPTION
**Pull request by dzhengfy**
Multiple PHB is supported in papar guest already. This is to test the migration
with multiple PHB between two PPC hosts.

Signed-off-by: Dan Zheng <dzheng@redhat.com>

### CODE REVIEW

File: libvirt/tests/src/migration/migrate_options_shared.py [hunk 1]
```diff
@@ -88,6 +89,20 @@ def check_virsh_command_and_option(command, option=None):
             test.cancel(msg + "virsh command '%s' with option '%s'" % (command,
                                                                        option))
 
+    def add_device(type="pci", index="0", model="pci-root"):
```
Comment (balamuruhans) [inline libvirt/tests/src/migration/migrate_options_shared.py]: type is a keyword, can we have different param name

File: libvirt/tests/src/migration/migrate_options_shared.py [hunk 2]
```diff
@@ -383,6 +402,25 @@ def check_timeout_postcopy(params):
...
+            if not remote_virsh_session:
+                remote_virsh_session = virsh.VirshPersistent(**remote_virsh_dargs)
+            target_guest_dumpxml = results_stdout_52lts(
+                remote_virsh_session.dumpxml(vm_name,
+                                             debug=True,
+                                             ignore_status=True)).strip()
+            check_content = xml_check_after_mig
+            if contrl_index:
+                all_ctrls = re.findall(check_content, target_guest_dumpxml)
+                if len(all_ctrls) != int(contrl_index) + 1:
```
Comment (balamuruhans) [inline libvirt/tests/src/migration/migrate_options_shared.py]: this should be `len(all_ctrls) != int(contrl_index)`
we iterate from `1 to int(contrl_index) + 1` but the total number would be `int(contrl_index)` right ?

File: libvirt/tests/src/migration/migrate_options_shared.py [hunk 3]
```diff
@@ -383,6 +402,25 @@ def check_timeout_postcopy(params):
...
             run_remote_cmd(cmd)
 
+        if xml_check_after_mig:
+            if not remote_virsh_session:
+                remote_virsh_session = virsh.VirshPersistent(**remote_virsh_dargs)
+            target_guest_dumpxml = results_stdout_52lts(
+                remote_virsh_session.dumpxml(vm_name,
+                                             debug=True,
+                                             ignore_status=True)).strip()
+            check_content = xml_check_after_mig
```
Comment (balamuruhans) [inline libvirt/tests/src/migration/migrate_options_shared.py]: we can directly use `xml_check_after_mig`, this `check_content` would still point the same addr location

File: libvirt/tests/src/migration/migrate_options_shared.py [hunk 4]
```diff
@@ -88,6 +89,20 @@ def check_virsh_command_and_option(command, option=None):
             test.cancel(msg + "virsh command '%s' with option '%s'" % (command,
                                                                        option))
 
+    def add_device(type="pci", index="0", model="pci-root"):
+        """
+        Add new device.
+        """
```
Comment (balamuruhans) [inline libvirt/tests/src/migration/migrate_options_shared.py]: I would appreciate to have the doc string for the method introduced.

**Comment by iccaszhulili** [comment]
rh-caselink retest please

**Comment by kylazhang** [comment]
@dzhengfy please fix the conflicts and comments

File: libvirt/tests/src/migration/migrate_options_shared.py (hunk 1 above)
Comment (dzhengfy) [reply libvirt/tests/src/migration/migrate_options_shared.py]: Updated

File: libvirt/tests/src/migration/migrate_options_shared.py (hunk 2 above)
Comment (dzhengfy) [reply libvirt/tests/src/migration/migrate_options_shared.py]: The total number of 'pci-root' includes the index 0, so it is int(contrl_index) + 1, thanks.

File: libvirt/tests/src/migration/migrate_options_shared.py (hunk 4 above)
Comment (dzhengfy) [reply libvirt/tests/src/migration/migrate_options_shared.py]: Docstring is added

File: libvirt/tests/src/migration/migrate_options_shared.py (hunk 3 above)
Comment (dzhengfy) [reply libvirt/tests/src/migration/migrate_options_shared.py]: updated.

**Comment by dzhengfy** [comment]
> @dzhengfy please fix the conflicts and comments

Fixed.

**Comment by dzhengfy** [comment]
```
# avocado run --vt-type libvirt virsh.migrate_options_shared.positive_test.multiple_phbs
JOB ID     : d6d48fe3568619918cf50474680c03bcd778b08a
JOB LOG    : /root/avocado/job-results/job-2018-09-25T05.16-d6d48fe/job.log
 (1/4) type_specific.io-github-autotest-libvirt.virsh.migrate_options_shared.positive_test.multiple_phbs.with_postcopy: PASS (233.72 s)
 (2/4) type_specific.io-github-autotest-libvirt.virsh.migrate_options_shared.positive_test.multiple_phbs.without_postcopy: PASS (224.66 s)
 (3/4) type_specific.io-github-autotest-libvirt.virsh.migrate_options_shared.positive_test.multiple_phbs.with_postcopy: PASS (225.02 s)
 (4/4) type_specific.io-github-autotest-libvirt.virsh.migrate_options_shared.positive_test.multiple_phbs.without_postcopy: PASS (224.87 s)
RESULTS    : PASS 4 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB TIME   : 965.61 s
```

File: libvirt/tests/src/migration/migrate_options_shared.py [hunk 5]
```diff
@@ -163,6 +164,24 @@ def check_virsh_command_and_option(command, option=None):
             test.cancel(msg + "virsh command '%s' with option '%s'" % (command,
                                                                        option))
 
+    def add_device(dev_type="pci", dev_index="0", dev_model="pci-root"):
+        """
+        Add new device
```
Comment (chunfuwen) [inline libvirt/tests/src/migration/migrate_options_shared.py]: 'Add new device' to 'Add multiple devices' sound more understandable

Comment (dzhengfy) [reply libvirt/tests/src/migration/migrate_options_shared.py]: Updated

File: libvirt/tests/src/migration/migrate_options_shared.py [hunk 6]
```diff
@@ -163,6 +164,24 @@ def check_virsh_command_and_option(command, option=None):
...
+
+        :param dev_type: the type of the device to be added
+        :param dev_index: the index of the device to be added
+        :param dev_model: the model of the device to be added
+        """
+        new_vm_xml = vm_xml.VMXML.new_from_inactive_dumpxml(vm_name)
+        for inx in range(1, int(dev_index) + 1):
```
[the remark this answers is not in the document]
Comment (dzhengfy) [reply libvirt/tests/src/migration/migrate_options_shared.py]: Remove all controllers. Updated.

**Pull request merged**